### ISSUE TITLE
[en] suppress_misspelled does not give the expected result

### ISSUE THREAD

**Issue by MikeUnwalla** [languagetool-org/languagetool#6295]
Refer to https://dev.languagetool.org/development-overview#the-basic-elements-of-a-rule

The archaic word 'mixt' gives a spelling error in all varieties of English. It has these postags: VBD, VBN.

The rule below should not suggest 'mixt', but it does. 'testrules' gives no error messages, the GUI gives the results that are shown in the examples, and https://community.languagetool.org/ruleEditor/expert?devMode=true tells me `Your example sentences work as expected.`

```
<rule id="SUPPRESS_MISSPELLED_TEST" name="suppress_misspelled test">
  <pattern>
    <token regexp="yes">blend|blends|blended</token>
  </pattern>
  <message>Use <suggestion suppress_misspelled="yes"><match no="1" postag_regexp="yes" postag="VB[PZDN]?">mix</match></suggestion>.</message>
  <example correction="Mix" type="incorrect"><marker>Blend</marker> the adhesive with the hardener.</example>
  <example correction="mixes" type="incorrect">The machine <marker>blends</marker> the paints.</example>
  <example correction="mixed|mixt" type="incorrect">The machine <marker>blended</marker> the paints.</example>
  <example correction="mixed|mixt" type="incorrect">The paints were <marker>blended</marker> carefully.</example>
  <example correction="mixed|mixt" type="incorrect">The <marker>blended</marker> mixture must be smooth.</example>
  <example type="correct"><marker>Mix</marker> the adhesive with the hardener.</example>
</rule>
```

**Comment by jaumeortola** [comment]
If "mixt" is in the synthesizer dictionary, `suppress_misspelled="yes"`will do nothing. 

Recently, we have created a filter that doesn't have this limitation. It checks the suggestions with the actual spelling rule. 
`<filter class="org.languagetool.rules.en.EnglishSuppressMisspelledSuggestionsFilter" args="suppressMatch:true"/>`
`suppressMatch:true` means that the whole match (and the message for the user) will be suppressed it there is no valid suggestion. 

Anyway, I added "mixt" to the file do-not-synthesize.txt, and that will solve the issue.
